crypto/dh/dh_kdf.c

```diff
@@ -33,20 +36,27 @@ int DH_KDF_X9_42(unsigned char *out, size_t outlen,
     if (oid_sn == NULL)
         return 0;
 
-    kdf = EVP_get_kdfbyname(SN_x942kdf);
-    if (kdf == NULL)
+    kdf = EVP_KDF_fetch(NULL, SN_x942kdf, NULL);
```

Comment:
The library context should probably be something else. Maybe the same as `md`'s?

Reply:
Okay, the MD properties will have been lost so it still isn't ideal.